lib/Runtime/Library/JavascriptSimdType.cpp

```diff
@@ -46,21 +46,22 @@ namespace Js
         AssertMsg(args.Info.Count > 0, "Should always have implicit 'this'");
         Assert(!(callInfo.Flags & CallFlags_New));
 
-        char16 buffer[512] = _u("");
-        char16* bufferPtr = buffer;
-
         if (args.Info.Count == 0 || !SIMDType::Is(args[0]))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedSimd, SIMDType::GetFullBuiltinName(&bufferPtr, _u("toLocaleString"))); //todo
+            char16 buffer[SIMD_STRING_BUFFER_MAX] = _u("");
+            swprintf_s(buffer, SIMD_STRING_BUFFER_MAX, _u("%s.toLocaleString()"), SIMDType::GetTypeName());
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedSimd, buffer);
         }
 
         //SPEC: The meanings of the optional parameters to this method are defined in the ECMA - 402 specification; 
         //implementations that do not include ECMA - 402 support must not use those parameter positions for anything else.
         //args[1] and args[2] are optional reserved parameters.
         RecyclableObject *obj = nullptr;
-        if (!JavascriptConversion::ToObject(args[0], scriptContext, &obj))  //Is this needed?
+        if (!JavascriptConversion::ToObject(args[0], scriptContext, &obj))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedSimd, SIMDType::GetFullBuiltinName(&bufferPtr, _u("toLocaleString")));
+            char16 buffer[SIMD_STRING_BUFFER_MAX] = _u("");
+            swprintf_s(buffer, SIMD_STRING_BUFFER_MAX, _u("%s.toLocaleString()"), SIMDType::GetTypeName());
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedSimd, buffer);
         }
```

Comment:
Minor nit: This block is a repetition of the above block.  Avoid duplication in codebase.  Fix it in future check-ins.